src/rpc/pulse/pulse.service.ts

```diff
@@ -0,0 +1,31 @@
+import {IncomingEvent} from "../../persistence/incoming-event/incoming-event.entity";
+import {IncomingEventLogType} from "../../persistence/incoming-event/incoming-event-log-type";
+import {msg} from "../../utils/logging/logging.service";
+import {IncomingEventService} from "../../persistence/incoming-event/incoming-event.service";
+import {Injectable} from "@nestjs/common";
+
+@Injectable()
+export class PulseService {
+
+  constructor(
+    private readonly incomingMessageService: IncomingEventService
+  ) {
+  }
+
+  handlePulse(pulseConfig: any, pulseHeader: any, pulseBody: any) {
+    let event = new IncomingEvent();
+    event.summary = pulseBody.summary;
+    event.eventTime = pulseHeader.eventTime;
+    event.serviceName = pulseHeader.service;
+    event.logType = pulseConfig.log ? IncomingEventLogType.ON : IncomingEventLogType.OFF;
+    event.ttl = pulseConfig.ttl;
+
+    this.incomingMessageService.create(event)
```

Comment:
we can actually use async and await  instead for .then for example

`try{
      await msg('incoming message saved ${this.incomingMessageService.create(event)}');
    }catch(error) {
      msg('error saving incoming message ${error}');
    } `

Reply:
yeah we can, but probably we do NOT want to wait until the database save operation finishes, as we want to return from this pulse method as quickly as possible. This will allow to avoid the database being a bottleneck when having a gazillion of pulse requests coming in.

Reply:
actually right now I am thinking that the function won't be blocked anyway, it's just a syntax sugar, so yeah I will apply that.

Reply:
Yah thats correct when we use async and await it looks like sequential programming and easy to understand if the code is too complex with multiple call backs